DMCompiler/DM/DMProc.cs

```diff
@@ -227,16 +227,30 @@ private class DMProcScope {
             WriteLabel(jumpTo);
         }
 
-        public void Break() {
-            Jump(_loopStack.Peek() + "_end");
+        public void Break(DMASTIdentifier label = null) {
+            if (label is not null)
+            {
+                Jump(label.Identifier + "_end");
+            }
+            else
+            {
+                Jump(_loopStack.Peek() + "_end");
+            }
         }
 
         public void BreakIfFalse() {
             JumpIfFalse(_loopStack.Peek() + "_end");
         }
 
-        public void Continue() {
-            Jump(_loopStack.Peek() + "_continue");
+        public void Continue(DMASTIdentifier label = null) {
+            if (label is not null)
+            {
+                Jump(label.Identifier + "_start");
```

Comment:
This doesn't continue a `for` loop properly. A `continue` needs to jump to a for's incrementor, which is at the end of the loop just before it jumps to the beginning.

A quirk to take note of is that label bodies can contain more than just a loop, and `continue label` will always continue the first loop in the body. `break label` will always exit the label body entirely.
```
var/i = 1
a:
	world.log << "A"
	for (,TRUE,i++)
		world.log << i
		break a
	world.log << "B"
sleep(10)
continue a
```
Outputs:
```
A
1
2
3
...
```